### DISCUSSION TITLE
Question about weight importance

### DISCUSSION THREAD

**Discussion by PRamoneda** [Raschka-research-group/coral-pytorch#28] [Q&A]
First of all, thank you so much for your work!

I am trying to use the coral loss function with an unbalanced dataset. Then, I suppose weight importance is for that however the input data have to be sized n_classes -1. I suppose class 0 can not be weighted. Am I right? 

However, class 0 is the smallest one! Do you have some suggestions?

Thank you so much!

**Comment by rasbt** [comment]
Ah, yes, good point. So the weight would be for the task, not the class label. Class 0 would still be weighted in a sense, since if you consider it as p(y > 0), the answer is Yes or No. So, the task weight here affects both class 0 and class 1.

**Comment by igorkf** [comment]
I was wondering if that's a good way of calculating importance weights for the tasks:
```
freq = df_val['label'].value_counts(normalize=True).sort_index()
importance_weights = torch.tensor([1 / freq.iloc[i:].sum() for i in range(1, len(freq))])
```

What do you guys think?
